include/nbl/asset/IDescriptorSetLayout.h

```diff
@@ -205,6 +314,8 @@ class NBL_API IDescriptorSetLayout : public virtual core::IReferenceCounted
                     static_assert(sizeof(size_t) == sizeof(bnd.samplers), "Bad reinterpret_cast!");
                     if (bnd.type == EDT_COMBINED_IMAGE_SAMPLER && bnd.samplers)
                         bnd.samplers = m_samplers->data() + reinterpret_cast<size_t>(bnd.samplers) - 1ull;
+
+					m_descriptorCount[bnd.type] += bnd.count;
                 }
 
                 // TODO: check for overlapping bindings (bad `SBinding` definitions)
```

Comment:
the `std::sort` below is not needed if you do https://github.com/Devsh-Graphics-Programming/Nabla/pull/345/files#r1047750723

Reply:
I'm not sure you need the AoS bindings anymore, so here's a way to counting sort them
```
for (auto it=_begin; it!=_end; it++)
{
    const binding_t* found = m_descriptorRedirects[it->type][binding_t{it->binding}];
    assert(found);
    m_descriptorRedirects.metadata.createFlags[std::distance(found)] = it->createFlags;
    m_descriptorRedirects.metadata.stageFlags[std::distance(found)] = it->stageFlags;
}
```
the samplers you shouldn't need to worry about when you ditch `SBindings`

or you could always slap the metadata into `SBindingRedirectBuildInfo` and not do this counting sort at all

Reply:
Resolved.